sdk/python/python.go

```diff
@@ -27,9 +27,51 @@ import (
 
 const windows = "windows"
 
+type CommandOpt interface {
+	apply(cmd *wrappedCmd)
+}
+
+type commandOptFunc func(cmd *wrappedCmd)
+
+func (c commandOptFunc) apply(cmd *wrappedCmd) {
+	c(cmd)
+}
+
+func WithDir(dir string) CommandOpt {
+	return commandOptFunc(func(cmd *wrappedCmd) {
+		cmd.cmd.Dir = dir
+	})
+}
+
+func WithEnv(env []string) CommandOpt {
+	return commandOptFunc(func(cmd *wrappedCmd) {
+		cmd.cmd.Env = env
+	})
+}
+
+func WithCommandPath(cmdPath string) CommandOpt {
+	return commandOptFunc(func(cmd *wrappedCmd) {
+		cmd.cmd.Path = cmdPath
+	})
+}
+
+func WithStdOut(stdout *os.File) CommandOpt {
+	return commandOptFunc(func(cmd *wrappedCmd) {
+		cmd.stdout = stdout
+		cmd.cmd.Stdout = stdout
+	})
+}
+
+func WithStdErr(stderr *os.File) CommandOpt {
+	return commandOptFunc(func(cmd *wrappedCmd) {
+		cmd.stderr = stderr
+		cmd.cmd.Stderr = stderr
+	})
+}
```

Comment:
I didn't exactly follow why we needed to add `CommandOpt` here - vs. just relying on string varargs as before?